apps/site/lib/site_web/controllers/schedule/line/diagram_helpers.ex

```diff
@@ -202,43 +223,59 @@ defmodule SiteWeb.ScheduleController.Line.DiagramHelpers do
   @spec build_branched_stop(
           RouteStop.t() | {RouteStop.t(), boolean},
           [stop_with_bubble_info],
-          {Route.branch_name(), [Route.branch_name()]}
+          {Route.branch_name(), [Route.branch_name()]},
+          boolean
         ) :: [stop_with_bubble_info]
-  def build_branched_stop(this_stop, all_stops, current_and_previous_branches)
+  def build_branched_stop(
+        this_stop,
+        all_stops,
+        current_and_previous_branches,
+        combine_green_branches
+      )
 
-  def build_branched_stop(stop, branch_stops, {_, ["Green" <> _ | _] = green_branches}) do
-    # Green Line always evaluates all branches on all stops. If the stop should have a bubble for a branch,
-    # &stop_bubble_type/3 returns a valid tuple, otherwise it returns false. The bubble list then gets filtered to
-    # remove anything that's not a tuple.
+  def build_branched_stop(
+        stop,
+        branch_stops,
+        {_, ["Green" <> _ | _] = green_branches},
+        combine_green_branches
```

Comment:
@digitalcora This breaks a few tests since it changes this function's arity -- what're your thoughts on making `combine_green_branches` an optional argument and/or giving it a default value of `false`? Alternately I can add the fourth argument to the relevant places in the tests and that works fine too.

Reply:
I could go either way. Follow your heart 💖